api/src/main/java/jakarta/enterprise/concurrent/ZonedTrigger.java

```diff
@@ -0,0 +1,163 @@
+ *              else if (hour{@literal >} 20)
+ *                  nextTime = nextTime.truncatedTo(ChronoUnit.DAYS)
+ *                                     .plusDays(day.equals(DayOfWeek.FRIDAY) ? 3 : 1)
+ *                                     .withHour(8);
+ *          }
+ *          return nextTime;
+ *      }
+ *  }
+ * </pre>
+ * <P>
+ *
+ * @since 3.0
+ */
+public interface ZonedTrigger extends Trigger {
+    /**
+     * Retrieve the next time that the task should run after.
+     *
+     * @param lastExecutionInfo information about the last execution of the task. 
+     *                          This value will be null if the task has not yet run.
+     * @param taskScheduledTime the date/time at which the
+     *                          {@code ManagedScheduledExecutorService.schedule}
+     *                          method was invoked to schedule the task.
+     * @return the date/time after which the next execution of the task should start.
+     */
+    ZonedDateTime getNextRunTime(LastExecution lastExecutionInfo, ZonedDateTime taskScheduledTime);
+
+    /**
+     * Retrieve the next time that the task should run after.
+     * <P>
+     * This method is provided to maintain compatibility with {@link Trigger} and should not be
+     * implemented. The default implementation delegates to the method signature that
+     * accepts and returns <code>ZonedDateTime</code>.
+     *
+     * @param lastExecutionInfo information about the last execution of the task. 
+     *                          This value will be null if the task has not yet run.
+     * @param taskScheduledTime the date/time at which the
+     *                          {@code ManagedScheduledExecutorService.schedule}
+     *                          method was invoked to schedule the task.
+     * @return the date/time after which the next execution of the task should start.
+     * @throws IllegalArgumentException if the next run time is too large to represent as a <code>Date</code>.
+     */
+    default Date getNextRunTime(LastExecution lastExecutionInfo, Date taskScheduledTime) {
+        ZonedDateTime nextTime = getNextRunTime(lastExecutionInfo, taskScheduledTime.toInstant().atZone(getZoneId()));
+        return Date.from(nextTime.toInstant());
+    }
+
+    /**
+     * Returns the timezone to use for the
+     * {@link java.time.ZonedDateTime ZonedDateTime} that is supplied to the
+     * {@link #getNextRunTime(LastExecution, java.time.ZonedDateTime) getNextRunTime} and
+     * {@link #skipRun(LastExecution, java.time.ZonedDateTime) skipRun} methods.
+     * <P>
+     * The default implementation returns the system default timezone
+     * and should be overridden whenever there is chance that the server
+     * might not be running with the same timezone for which the
+     * business logic within this trigger is written.
+     *
+     * @return timezone to use for operations on this trigger.
+     */
+    default ZoneId getZoneId() {
```

Comment:
does this need to be public?

Reply:
The interface is public, so these methods are implicitly public.  I'd be happy to explicitly add `public` to it for clarity.  Here is a commit that updates it:
https://github.com/eclipse-ee4j/concurrency-api/pull/143/commits/25ea34982ad306bfef77dc697c8e1fd6ce4ad48a